hbw/app/javascript/packs/hbw/components/shared/context/error_handling.tsx

```diff
@@ -0,0 +1,90 @@
+import Messenger from 'messenger';
+import React, {
+  createContext, ComponentType, FC, useState, useCallback, useContext,
+} from 'react';
+import HBWError from '../../error';
+import TranslationContext, { ContextType as TranslationContextType } from './translation';
+import ConnectionContext, { ContextType as ConnectionContextType } from './connection';
+import { CamundaError, WidgetErrorType } from '../utils/errors';
+
+export type ContextType = {
+  addError: Function,
+  formError: JSX.Element,
+  safeRequest: Function
+}
+
+const ErrorHandlingContext = createContext<ContextType | null>(null);
+
+export const withErrorHandlingContext = () => (WrappedComponent: ComponentType) => {
+  const ErrorHandlingProvider: FC = (props) => {
+    const { translate: t } = useContext(TranslationContext) as TranslationContextType;
+    const { request } = useContext(ConnectionContext) as ConnectionContextType;
+
+    const [errorStack, setErrorStack] = useState<Array<WidgetErrorType>>(() => []);
+    const [formError, setFormError] = useState<JSX.Element>(() => <HBWError/>);
+
+    const notify = useCallback((error: WidgetErrorType) => {
+      if (error.level === 'warn') {
+        // @ts-ignore
+        Messenger.warn(error.notificationText);
+      } else if (error.level === 'alert') {
+        // @ts-ignore
+        Messenger.alert(error.notificationText);
+      } else if (error.level === 'error') {
+        // @ts-ignore
+        Messenger.error(error.notificationText);
+      }
+    }, []);
+
+    const setupFormError = useCallback((error: WidgetErrorType) => {
+      setFormError(
+        <HBWError
+          errorHeader={error.header}
+          errorBody={error.body}
+        />,
+      );
+    }, [formError, errorStack]);
+
+    const translateError = useCallback((error: WidgetErrorType) => {
+      error.header = t(error.header, {}, error.header);
+      error.notificationText = error.notificationText ? t(error.notificationText, {}, error.notificationText) : '';
+    }, []);
+
+    const addError = useCallback((error: WidgetErrorType) => {
+      translateError(error);
+
+      if (!errorStack.map(value => value.header).includes(error.header)) {
+        errorStack.push(error);
+        setErrorStack(errorStack);
```

Comment:
Why do you think we should keep errors in some stack instead of just raising it? You aren't using it anywhere.

Reply:
Also, pretend we got the same error the second time. Why do we shouldn't raise it?